server/sonar-server/src/test/java/org/sonar/server/qualitygate/RegisterQualityGatesTest.java

```diff
@@ -52,32 +63,165 @@
 
   @Rule
   public DbTester db = DbTester.create(System2.INSTANCE);
+  @Rule
+  public LogTester logTester = new LogTester();
+
+  private static final String A_RATING_VALUE = Integer.toString(RatingGrid.Rating.A.getIndex());
+  private static final int LEAK_PERIOD = 1;
+  private static final String BUILT_IN_NAME = "Sonar way";
+
+  private DbClient dbClient = db.getDbClient();
+  private DbSession dbSession = db.getSession();
+
+  private QualityGateDao qualityGateDao = dbClient.qualityGateDao();
+  private QualityGateConditionDao gateConditionDao = dbClient.gateConditionDao();
+  private MetricDao metricDao = dbClient.metricDao();
+  private QualityGateConditionsUpdater qualityGateConditionsUpdater = new QualityGateConditionsUpdater(dbClient);
 
-  DbClient dbClient = db.getDbClient();
-  DbSession dbSession = db.getSession();
+  private QualityGates qualityGates = mock(QualityGates.class);
```

Comment:
It's strange to use real DAO but mock for QualityGates => I would recommend to remove usage of QualityGates (which should be removed, have a look at the javadoc. But I agree, it should be deprecated to better know we should not use it !)

Reply:
That's the line I did not touch on this code. 
The mock is used to check that there is an interaction with setDefault (like it was previously). I would be happy to remove this.